### PR TITLE
Fix discard_stale (Closes #65)

### LINKED ISSUE #65: Stale items are not ignored with discard_stale: true when no status has been published at all
With a configuration like this
```
analyzers:
  my_stale_item:
    type: diagnostic_aggregator/GenericAnalyzer
    path: Nonexistent1
    find_and_remove_prefix: 'nonexistent1'
  my_stale_item_that_should_be_ignored:
    type: diagnostic_aggregator/GenericAnalyzer
    path: Nonexistent2
    find_and_remove_prefix: 'nonexistent2'
    discard_stale: true
    timeout: 5.0
```
and with no messages published on `/diagnostics`, I would expect "Nonexistent1" to show up as stale and "Nonexistent2" to *not* show up in `/diagnostics_agg`.

However, both are always present in `/diagnostics_agg`.
Here's a minimal launch file to reproduce:
```XML
<launch>

  <node pkg="diagnostic_aggregator" type="aggregator_node" name="diagnostic_aggregator">
    <rosparam>
      analyzers:
        my_stale_item:
          type: diagnostic_aggregator/GenericAnalyzer
          path: Nonexistent1
          find_and_remove_prefix: 'nonexistent1'
        my_stale_item_that_should_be_ignored:
          type: diagnostic_aggregator/GenericAnalyzer
          path: Nonexistent2
          find_and_remove_prefix: 'nonexistent2'
          discard_stale: true
          timeout: 5.0
    </rosparam>
  </node>

  <node name="$(anon monitor)" pkg="rqt_robot_monitor" type="rqt_robot_monitor"/>

</launch>
```

### PR DESCRIPTION
**Pull request by g-gemignani**
Make sure that analyzers flagged with discard_stale = true are correctly
removed after being stale for a period greater than the timeout

### CODE REVIEW

File: diagnostic_aggregator/test/discard_stale_not_published_test.py [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+        sub_agg = rospy.Subscriber("/diagnostics_agg", DiagnosticArray, self.diag_agg_cb)
+        self._start_time = rospy.Time.now()
+
+    def diag_agg_cb(self, msg):
+        with self._mutex:
+            self._agg_expecteds = []
+            for stat in msg.status:
+                self._agg_expecteds.append(stat)
+
+    def test_discard_stale(self):
```
Comment (trainman419) [inline diagnostic_aggregator/test/discard_stale_not_published_test.py:75]: In order to properly test the original bug, this test should publish some diagnostic items with the `nonexistent2` prefix, and then wait for them to go stale and be removed before checking that they're no longer published by the diagnostic aggregator.

Comment (g-gemignani) [reply diagnostic_aggregator/test/discard_stale_not_published_test.py:75]: addressed the issue

**Comment by g-gemignani** [comment]
Hello,
thank you for the feedback on the pr. Indeed I found a problem with it, where actually the timeout period of stale time was skipped. I addressed the issue and updated the tests.
Thank you!

**Comment by trainman419** [comment]
Thanks!

**Pull request merged**